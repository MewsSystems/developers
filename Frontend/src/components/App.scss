html,
body {
  margin: 0;
  padding: 0;
}

html {
  background: #000;
  color: #fff;
}

// react-paginate library doesn't provide api to replace its componets
// so I didn't find a way to use styled components with it
.Search-Page-Pagination {
  display: flex;
  list-style: none;
  justify-content: center;
  margin: 2rem 0;

  .previous,
  .next {
    display: none;
  }

  li {
    border: 1px solid #fff;
    margin: 0 0.25rem;
    padding: 0;
    cursor: pointer;

    &.break {
      border: none;
    }

    &.selected {
      background: #fff;
      color: #000;
      border-color: #000;
    }
  }

  a {
    display: block;
    padding: 0.5rem;
  }
}
