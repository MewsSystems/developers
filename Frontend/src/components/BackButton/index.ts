export { default } from './BackButton';
export * from './styled';
