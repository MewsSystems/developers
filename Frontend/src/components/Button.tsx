import React from 'react'
import styled, { css } from 'styled-components'
import { BORDER_RADIUS, BOX_SHADOW } from 'constants/index'
import { COLORS } from 'constants/colors'
import { darken } from 'polished'

export const StyledButton = styled.button<
  Pick<ButtonProps, 'disabled' | 'variant' | 'fullWidth'>
>`
  width: ${({ fullWidth }) => fullWidth && '100%'};
  min-width: 12rem;
  padding: 1rem;
  text-align: center;
  text-transform: uppercase;
  border: 0;
  border-radius: ${BORDER_RADIUS.MEDIUM};
  box-shadow: ${BOX_SHADOW.MEDIUM};

  ${({ disabled, variant }) =>
    disabled
      ? css`
          background: ${COLORS.GRAY};
          color: ${COLORS.DARK_GRAY};
        `
      : css`
          background: ${variant === 'primary' ? COLORS.BLUE : COLORS.RED};
          color: ${COLORS.WHITE};
          cursor: pointer;
          transition: background ease-in 0.5s;

          :hover {
            background: ${variant === 'primary'
              ? darken(0.1, COLORS.BLUE)
              : darken(0.1, COLORS.RED)};
            transition: background ease-in 0.5s;
          }
        `}
`

export interface ButtonProps {
  variant?: 'primary' | 'secondary'
  disabled?: boolean
  fullWidth?: boolean
  onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void
  children?: React.ReactNode | React.ReactNodeArray
}

export const Button: React.FC<ButtonProps> = ({
  variant = 'primary',
  disabled,
  fullWidth,
  onClick,
  children,
}) => (
  <StyledButton
    variant={variant}
    disabled={disabled}
    fullWidth={fullWidth}
    onClick={onClick}
  >
    {children}
  </StyledButton>
)
