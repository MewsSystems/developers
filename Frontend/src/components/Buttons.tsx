import styled from '../utils/styled'

const ThemeSwitcherButton = styled('button')`
  display: inline-block;
  padding: 0.25rem 0.5rem;
  border: 1px solid ${props => props.theme.colors.white};
  border-radius: 3px;
  background-color: ${props => props.theme.colors.white};
  color: ${props => props.theme.colors.brand};
  font-size: 0.8rem;
  text-transform: uppercase;
  letter-spacing: 1px;
  cursor: pointer;
  transition: all 0.3s ease;

  &:hover,
  &:focus {
    background-color: transparent;
    color: ${props => props.theme.colors.white};
  }
`

export default ThemeSwitcherButton
