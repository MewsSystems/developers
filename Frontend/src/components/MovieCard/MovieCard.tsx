import { Link, LinkProps } from 'react-router-dom';
import { SearchMovieItem } from '../../services/tmdbApi';
import { splitDate } from '../../utils';
import MoviePoster from '../MoviePoster';
import { Card, CardBody, CardText, CardTitle, CardProps } from './styled';

interface PlainMovieCardProps {
  isLink?: false;
}

interface LinkMovieCardProps {
  isLink: true;
  to: LinkProps['to'];
}

const MovieCard = (
  props: SearchMovieItem &
    CardProps &
    (PlainMovieCardProps | LinkMovieCardProps)
) => {
  const cardProps = props.isLink ? { $as: Link, to: props.to } : {};
  const [releaseYear] = splitDate(props.release_date);
  const details = [
    releaseYear || 'N/A',
    props.original_language.toUpperCase(),
    `${props.vote_average}/10`,
  ];

  return (
    <Card {...cardProps} height={props.height} width={props.width}>
      <MoviePoster
        posterPath={props.poster_path}
        alt={props.title}
        width="92px"
        height={'inherit'}
      />
      <CardBody>
        <CardTitle>
          {props.title}{' '}
          {props.title !== props.original_title && (
            <small>({props.original_title})</small>
          )}
        </CardTitle>
        <div>{details.join(' \u25CF ')}</div>
        <CardText>{props.overview}</CardText>
      </CardBody>
    </Card>
  );
};

export default MovieCard;
