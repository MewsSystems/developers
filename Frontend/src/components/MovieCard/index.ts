export { default } from './MovieCard';
export * from './styled';
