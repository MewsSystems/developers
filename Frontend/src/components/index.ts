import styled from "styled-components";
import { Link } from "react-router-dom";

export const Layout = styled.div`
  display: grid;
  grid-template-columns: 350px 1fr 350px;
  height: 100vh;
`;

export const CenterColumn = styled.div`
  grid-column: 2;
  padding: 20px;
`;

export const Card = styled.div`
  box-shadow: 0 8px 16px 0 rgba(0,0,0,0.2);
  display: flex;
  padding: 10px;
  color: gray;
  font: 16px/1.5 Helvetica, Verdana, sans-serif;
`

export const Content = styled.div`
  padding: 2px 16px;
`

export const Title = styled.div`
  display: block;
  margin: 10px 0;
  font-weight: bold;
  color: black;
  font-size:20px;
`;

export const SubTitle = styled.div`
  font-size:14px;
  color:grey;
  margin:2px 0;
`;

export const Paragraph = styled.p`
  line-height: 18px;
  margin: 5px;
`;

export const ErrorMessage = styled.div`
  border:1px solid red;
  padding:10px;
  color:red;
  font: 18px Helvetica, Verdana, sans-serif;
`



export const StyledInput = styled.input`
  background-color: transparent;
  border: none;
  border-bottom: 1px solid #9e9e9e;
  border-radius: 0;
  outline: none;
  height: 3rem;
  width: 100%;
  font-size: 16px;
  margin: 0 0 8px 0;
  padding: 0;
  box-shadow: none;
  box-sizing: content-box;
  transition: box-shadow .3s, border .3s;
`;

export const StyledLabel = styled.label`
  width:100%;
  color: #9e9e9e;
  position: absolute;
  top: -15px;
  left: 0;
  font-size: 1rem;
  cursor: text;
  transition: transform .2s ease-out, color .2s ease-out;
  text-align: initial;
  &::after {
    display: block;
    content: "";
    position: absolute;
    top: 100%;
    left: 0;
    opacity: 0;
    transition: .2s opacity ease-out, .2s color ease-out;
  }
`

export const Button = styled.button`
  text-decoration: none;
  color: #fff;
  background-color: #26a69a;
  text-align: center;
  letter-spacing: .5px;
  transition: background-color .2s ease-out;
  cursor: pointer;
  font-size: 14px;
  outline: 0;
  border: none;
  border-radius: 2px;
  display: inline-block;
  height: 36px;
  line-height: 36px;
  padding: 0 16px;
  text-transform: uppercase;
  vertical-align: middle;
  &:disabled {
    background-color:#ccc;
  }
`;

export const StyledLink = styled(Link)`
  text-decoration: none;
  color: #26a69a;
`;
