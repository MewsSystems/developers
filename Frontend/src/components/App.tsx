import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Search from './Search/Search';
import Movie from './Movie/Movie';
import { Provider } from 'react-redux';
import createStore from '../store';
import './App.scss';

const store = createStore();

const App = () => {
  return (
    <Provider store={store}>
      <Router>
        <Switch>
          <Route path="/" exact component={Search} />
          <Route path="/movie/:id" render={({ match }) => <Movie id={match.params.id} />} />
        </Switch>
      </Router>
    </Provider>
  );
};

export default App;
