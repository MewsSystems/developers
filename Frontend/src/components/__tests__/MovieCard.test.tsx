import { MovieCard } from 'components/MovieCard'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import React from 'react'

describe('Test Card component', () => {
  it('Tests snapshot', () => {
    expect(
      shallow(
        <MovieCard
          id={1}
          background="test.jpg"
          language="en"
          overview="test"
          title="test"
        />
      )
    ).toMatchSnapshot()
  })
})
