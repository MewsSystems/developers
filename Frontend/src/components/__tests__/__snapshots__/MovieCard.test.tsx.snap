// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card component Tests snapshot 1`] = `
<styled.div
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "linear-gradient(180deg, rgba(0, 0, 0, 0) 0%, rgba(0, 0, 0, 0.2) 55%, rgba(0, 0, 0, 0.5) 100%), url(test.jpg)",
    }
  }
>
  <styled.div>
    <styled.h2>
      test
    </styled.h2>
    <styled.span>
      en
    </styled.span>
  </styled.div>
  <styled.p>
    test
  </styled.p>
</styled.div>
`;
